### ISSUE TITLE
Boxed closure in struct: trait bound is not satisfied

### ISSUE THREAD

**Issue by kzdnk** [loganmzz/macon-rs#29]
Is it possible to derive a builder for a struct with one field being a boxed closure?

e.g.
```rust
#[derive(macon::Builder)]
struct Test {
    f: Box<dyn Fn(usize) -> usize>,
}

fn main() {
    // works
    Test {
        f: Box::new(|x| x + 1),
    }; 
    // fails
    Test::builder().f(Box::new(|x| x + 1));
}
``` 
results in 
```
macon-boxed-fn [ master][?][RUST 1.79.0]
:> cargo b
   Compiling macon-boxed-fn v0.1.0 (/mnt/data/projects/other/macon-boxed-fn)
error[E0277]: the trait bound `Box<(dyn Fn(usize) -> usize + 'static)>: From<Box<{closure@src/main.rs:11:32: 11:35}>>` is not satisfied
  --> src/main.rs:11:23
   |
11 |     Test::builder().f(Box::new(|x| x + 1));
   |                     - ^^^^^^^^^^^^^^^^^^^ the trait `From<Box<{closure@src/main.rs:11:32: 11:35}>>` is not implemented for `Box<(dyn Fn(usize) -> usize + 'static)>`, which is required by `Box<{closure@src/main.rs:11:32: 11:35}>: Into<Box<(dyn Fn(usize) -> usize + 'static)>>`
   |                     |
   |                     required by a bound introduced by this call
   |
   = help: the following other types implement trait `From<T>`:
             <Box<(dyn std::error::Error + 'a)> as From<&str>>
             <Box<(dyn std::error::Error + 'a)> as From<Cow<'b, str>>>
             <Box<(dyn std::error::Error + 'a)> as From<E>>
             <Box<(dyn std::error::Error + 'a)> as From<String>>
             <Box<(dyn std::error::Error + Send + Sync + 'a)> as From<&str>>
             <Box<(dyn std::error::Error + Send + Sync + 'a)> as From<Cow<'b, str>>>
             <Box<(dyn std::error::Error + Send + Sync + 'a)> as From<E>>
             <Box<(dyn std::error::Error + Send + Sync + 'a)> as From<String>>
           and 18 others
   = note: required for `Box<{closure@src/main.rs:11:32: 11:35}>` to implement `Into<Box<(dyn Fn(usize) -> usize + 'static)>>`
note: required by a bound in `TestBuilder::f`
  --> src/main.rs:1:10
   |
1  | #[derive(macon::Builder)]
   |          ^^^^^^^^^^^^^^ required by this bound in `TestBuilder::f`
2  | struct Test {
3  |     f: Box<dyn Fn(usize) -> usize>,
   |     - required by a bound in this associated function
   = note: this error originates in the derive macro `macon::Builder` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `macon-boxed-fn` (bin "macon-boxed-fn") due to 1 previous error

```

**Comment by loganmzz** [comment]
Hi @ziutech,

Thank you for taking time to play with `Macon` :)

I need to investigate further in order to fine the best solution. As a quick win, I suggest to add [field/struct setting](https://docs.rs/macon/latest/macon/#settings-1) in order to disable `Into` support for setter?
I will check if it works and publish a new release with this new feature.

**Comment by loganmzz** [comment]
Hi @ziutech,

Sorry for the delay, I finally released [1.1.0](https://crates.io/crates/macon/1.1.0) which should help fixing your issue. See [`Into` argument section](https://docs.rs/macon/1.1.0/macon/#into-argument) for the details.

In short:

```rust
#[derive(Builder)]
struct Foo {
  #[builder(Into=!)]
  dyn_trait: Box<dyn Fn(usize)->usize>,
}
```

**Comment by kzdnk** [comment]
> Hi @ziutech,
> 
> Sorry for the delay, I finally released [1.1.0](https://crates.io/crates/macon/1.1.0) which should help fixing your issue. See [`Into` argument section](https://docs.rs/macon/1.1.0/macon/#into-argument) for the details.
> 
> In short:
> 
> ```rust
> #[derive(Builder)]
> struct Foo {
>   #[builder(Into=!)]
>   dyn_trait: Box<dyn Fn(usize)->usize>,
> }
> ```

That's great! Thank you!
